demos/CMakeLists.txt

```diff
@@ -37,6 +37,16 @@ target_link_libraries(spectre_v4 safeside)
 add_executable(ret2spec_cyclic ret2spec_cyclic.cc)
 target_link_libraries(ret2spec_cyclic safeside)
 
+# Spectre V2 -- branch target injection
+add_executable(spectre_v2 spectre_v2.cc)
+if((${CMAKE_SYSTEM_NAME} MATCHES "^Darwin$"))
+  target_compile_options(spectre_v2 PRIVATE -fomit-frame-pointer)
```

Comment:
Why do you need this flag?

Reply:
I need to avoid pushing rbp on the stack and popping it from the stack. Since the jump target is shifted instruction by instruction, I need to avoid that prologue that is otherwise created by using frame pointers. MacOS LLVM is specific from the point of view that -O2 does not imply -fomit-frame-pointer as on other compilers.